addons/l10n_it_edi/models/account_invoice.py

```diff
@@ -157,21 +157,35 @@ def format_alphanumeric(text_to_convert):
                 if tax.amount == 0.0:
                     tax_map[tax] = tax_map.get(tax, 0.0) + line.price_subtotal
 
+        reverse_charge = document_type in ['TD17', 'TD18', 'TD19']
+        # Represent if the document is a reverse charge refund in a single variable
+        rc_refund = self.move_type == 'in_refund' and reverse_charge
+
         # Constraints within the edi make local rounding on price included taxes a problem.
         # To solve this there is a <Arrotondamento> or 'rounding' field, such that:
         #   taxable base = sum(taxable base for each unit) + Arrotondamento
         tax_details = self._prepare_edi_tax_details(
             filter_to_apply=lambda l: l['tax_repartition_line_id'].factor_percent >= 0
         )
         for _tax_name, tax_dict in tax_details['tax_details'].items():
-            base_amount = tax_dict['base_amount_currency']
-            tax_amount = tax_dict['tax_amount_currency']
+            # Currency can be must be in euros for reverse charge documents
+            if self.currency_id.name == 'EUR' or not reverse_charge:
+                base_amount = tax_dict['base_amount_currency']
+                tax_amount = tax_dict['tax_amount_currency']
+            else:
+                # The assumption is that the company currency is EUR. Making a reverse charge document
+                # requires a company is Italian, and if a company is Italian then it's leagaly required
+                # to use EUR.
+                base_amount = tax_dict['base_amount_currency'] = tax_dict['base_amount']
```

Comment:
Is not it weird to hack base_amount_currency?  It might also mess it up for people not updating the module.  @dbkosky